vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java

```diff
@@ -0,0 +1,68 @@
+package edu.asu.diging.vspace.web.staff.api;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.IVSImage;
+import edu.asu.diging.vspace.core.services.IImageService;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Controller
+public class ImagesSearchFullApiController {
+
+    @Autowired
+	private IImageService imageService;
+
+    @Autowired
+	private ISpaceManager spaceManager;
+
+    @RequestMapping("/staff/images/search/full")
+	public String imageSearchFull(Model model,
+	@RequestParam(value = "searchText", required = false) String searchTerm) {
+    	return String.format("redirect:/staff/images/search/full/1?searchText=%s");
+    }
+    
+    @RequestMapping("/staff/images/search/full/{page}")
+	public String imageSeacrhDescription(@PathVariable(required = false) String page,
+			@RequestParam(value = "searchText", required = false) String searchTerm, Model model,
+			RedirectAttributes attributes) {
+    	List<IVSImage> images = null;
+    	if (searchTerm != null && !searchTerm.trim().isEmpty()) {
+            images = imageService.search(searchTerm);
+    	}
+    	else {
+            images = imageService.getImages(1);
+    	}
+    	int pageNo;
+    	page = StringUtils.isEmpty(page) ? "1" : page;
+    	try {
+            pageNo = imageService.validatePageNumber(Integer.parseInt(page), images);
```

Comment:
the page object that the repository returns when you search should have all the info you need (page number, total pages, etc). no reason to make extra calls to the db here.